packages/core/core/src/PackagerRunner.js

```diff
@@ -10,20 +11,23 @@ class PackagerRunner {
     this.dirExists = false;
   }
 
-  async loadPackager() {
-    return require('@parcel/packager-js');
-  }
+  async runPackager(bundle) {
+    let {packagers} = this.parcelConfig;
+    let {name = 'file.' + bundle.type} = bundle;
+    let packager = matchConfig(packagers, name);
+
+    if (!packager) {
+      throw new Error(
+        `Could not find packager for bundle of type "${bundle.type}"`
+      );
+    }
 
-  async runPackager({bundle}) {
-    let packager = await this.loadPackager();
+    // TODO(fathyb): use ConfigRunner
+    packager = require(packager);
 
     let modulesContents = await Promise.all(
       bundle.assets.map(async asset => {
-        // let fileContents = await packager.readFile({
-        //   filePath: asset.code,
-        // });
         let blobs = await this.cache.readBlobs(asset);
-
         let result = await packager.asset({blobs});
```

Comment:
I think I'd like to change the API to only have one method on a packager: `package`, which receives a list of assets with their output pre-read from the cache. Packagers also shouldn't have any filesystem related methods since the optimizers still need to run after the packagers.